flink-runtime/src/test/java/org/apache/flink/runtime/rest/handler/legacy/metrics/MetricFetcherTest.java

```diff
@@ -184,4 +179,52 @@ public String getValue() {
 
 		return dump;
 	}
+
+	@Test
+	public void testLongUpdateInterval() {
+		final long updateInterval = 1000L;
+		final AtomicInteger requestMetricQueryServicePathsCounter = new AtomicInteger(0);
+		final RestfulGateway restfulGateway = createRestfulGateway(requestMetricQueryServicePathsCounter);
+
+		final MetricFetcher fetcher = createMetricFetcher(updateInterval, restfulGateway);
+
+		fetcher.update();
+		fetcher.update();
+
+		assertThat(requestMetricQueryServicePathsCounter.get(), is(1));
+	}
+
+	@Test
+	public void testShortUpdateInterval() throws InterruptedException {
```

Comment:
I have doubts regarding the stability of this test since the MetricFetcher internally relies on `System.currentTimeMillis`, which isn't necessarily as granular as this test requires. I would increase the sleep to 50.

Reply:
Alright, I've hardened the test case with
```
final long start = System.currentTimeMillis();
		long difference = 0L;

		while (difference <= updateInterval) {
			Thread.sleep(2L * updateInterval);
			difference = System.currentTimeMillis() - start;
		}
```